src/feature/relay/router.c

```diff
@@ -865,15 +872,17 @@ router_write_fingerprint(int hashed)
   cp = read_file_to_str(keydir, RFTS_IGNORE_MISSING, NULL);
   if (!cp || strcmp(cp, fingerprint_line)) {
     if (write_str_to_file(keydir, fingerprint_line, 0)) {
-      log_err(LD_FS, "Error writing %sfingerprint line to file",
-              hashed ? "hashed " : "");
+      log_err(LD_FS, "Error writing %s%s line to file",
+              hashed ? "hashed " : "",
+              ed25519_identity ? "ed25519 identity" : "fingerprint");
       goto done;
     }
   }
 
-  log_notice(LD_GENERAL, "Your Tor %s identity key fingerprint is '%s %s'",
-             hashed ? "bridge's hashed" : "server's", options->Nickname,
-             fingerprint);
+  log_notice(LD_GENERAL, "Your Tor %s identity key %s is '%s %s'",
+             hashed ? "bridge's hashed" : "server's",
+             ed25519_identity ? "ed25519 key" : "fingerprint",
+             options->Nickname, fingerprint);
```

Comment:
Small problem here. Log output looks like:

`
Jun 25 08:06:41.194 [notice] Your Tor server's identity key ed25519 key is 'mytest xyfc3udfBrGnhD8i+4bmbdN5QMBNfZ9Xyl1aR/6dYcY'
`

I would probably go for: `Your Tor server's identity key ed25519 fingerprint is ...`